src/Sentry/Internal/Hub.cs

```diff
@@ -26,12 +26,15 @@ public Hub(SentryOptions options)
 
             if (options.Dsn == null)
             {
-                if (!Dsn.TryParse(DsnLocator.FindDsnStringOrDisable(), out var dsn))
+                var dsn = Dsn.TryParse(DsnLocator.FindDsnStringOrDisable());
```

Comment:
What about we call it just `Parse` then? Since `TryParse` feels like the classic bool returning `out` parameter approach.

Reply:
Up to you. The current `Parse` works differently, i.e. it throws with a descriptive error. I think that could still be useful.

> Since TryParse feels like the classic bool returning out parameter approach.

I think this will pass :)
I had it at first and then got used to it, now I never use `out` for this.

Reply:
It's a .NET convention that has existed for 20 years though, it's what people expect.

Since this is internal API it doesn't really matter. But I'd vote for a simple `Parse` or call it `ParseOrNull` or something descriptive.

Since it's internal the second method that throws isn't needed, we can have a single one, maybe one that throws and we handle it?

![image](https://user-images.githubusercontent.com/1633368/94465808-767fc180-018e-11eb-8500-9031de9348e9.png)
Reading this I assume `dsn` is bool.

Reply:
Can you consider the difficulty of changing it to internal? I mentioned in the PR description.

Reply:
I thought I did with this comment:

> We could revisit those callbacks, maybe get rid of them or replace with something else. 
Lets discuss what problem are we solving with them and how else could this be addressed?

Reply:
Maybe we just pass SentryOptions to the callback. That give more control to the caller to decide how to create the the HttpClient.

Still up for discussion what hooks we should put there. I added those without a lot (any) of input from customers

Reply:
> Maybe we just pass SentryOptions to the callback.

I like this way more, yes.

Reply:
But then again, those callbacks are probably set from the context of initializing the SDK anyway. So something like this:

```csharp
SentrySdk.Init(o =>
{
    // SentryOptions (o) is already in scope here so no reason to pass it as a parameter
    o.CreateHttpClientHandler = () => new HttpClientHandler();
});
```

Reply:
So maybe just simplify the signature by removing `Dsn` altogether?